src/linear_algebra/finsupp.lean

```diff
@@ -217,13 +217,18 @@ begin
   exact linear_map.is_linear _
 end
 
-def lsum (f : α → R →ₗ[R] M) : (α →₀ R) →ₗ[R] M :=
+/-- finsupp.sum as a linear map. -/
+def lsum (f : α → M →ₗ[R] N) : (α →₀ M) →ₗ[R] N :=
 ⟨λ d, d.sum (λ i, f i),
   assume d₁ d₂, by simp [sum_add_index],
-  assume a d, by simp [sum_smul_index, smul_sum, -smul_eq_mul, smul_eq_mul.symm]⟩
+  assume a d, by simp [sum_smul_index', smul_sum]⟩
+
```

Comment:
You could add a simp lemma for `coe_lsum` that simps away the coe to fun.

Reply:
That's `lsum_apply`.

Reply:
No, not exactly. Some time ago @urkud  pointed out that it might actually be better to have simp lemmas that act solely on the coerced function, even before it is applied to some term, and then simp it to a lambda term.

Reply:
Done